src/radical/pilot/agent/agent_0.py

```diff
@@ -82,8 +84,12 @@ def __init__(self, cfg, session):
         self._cmgr.start_bridges()
         self._cmgr.start_components()
 
-        # start any services if they are requested
-        self._start_services()
+        # uid of service tasks
+        self._service_task_ids = []
+        # Event to handle for services started
+        self.services_event = threading.Event()
+        # currently running services
+        self.running_services = []
```

Comment:
Comments should be meaningful and provide information which is not obvious.  This is not a good comment:
```py
self.running_services = ...  # running services
```
The purpose is clear in the var name (as it should be).

Please use a leading underscore for variables which are private.

Reply:
added _